source/distributed/fully_distributed_tria.cc

```diff
@@ -285,6 +289,98 @@ namespace parallel
 
 
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::copy_triangulation(
+      const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      Assert(this->dynamic_construction_data,
+             ExcMessage("No DynamicConstructionData has been provided!"));
+
+      // temporal array to store user subdomain_ids and level_subdomain_ids
+      // (since these might be changed during partitioning)
+      std::vector<std::pair<types::subdomain_id, types::subdomain_id>>
+        stored_cell_subdomain_ids;
+
+      if (dynamic_cast<const dealii::parallel::TriangulationBase<dim, spacedim>
+                         *>(&other_tria) == nullptr)
+        {
+          // check if other triangulation is not parallel triangulation;
+          // this means we have a serial triangulation, which has to be
+          // partitioned!
+
+          auto &serial_tria =
+            *const_cast<dealii::Triangulation<dim, spacedim> *>(&other_tria);
```

Comment:
It seems that you need the `const_cast` to be compatible with the various `GridTools::partition` functions. From an interface perspective, would it make sense to either copy the triangulation here? I know that you restore the old state in the end, but the question is whether the memory consumption is significant. If yes, please add a comment along those lines ("we rather overwrite the given triangulation's subdomain ids here than create a more memory-heavy copy").